picoquic/logger.c

```diff
@@ -577,60 +580,44 @@ size_t picoquic_log_connection_close_frame(FILE* F, uint8_t* bytes, size_t bytes
+        fprintf(F, "    Malformed %s, requires %d bytes out of %d\n",
+            picoquic_log_frame_names(ftype), 
             (int)(byte_index + picoquic_varint_skip(bytes + 3)), (int)bytes_max);
-        return bytes_max;
-    } else {
-        byte_index += l1;
+        byte_index = bytes_max;
     }
+    else {
+        byte_index += l1;
 
-    fprintf(F, "    CONNECTION CLOSE, Error 0x%04x, Reason length %llu\n",
-        error_code, (unsigned long long)string_length);
-    if (byte_index + string_length > bytes_max) {
-        fprintf(F, "    Malformed CONNECTION CLOSE, requires %llu bytes out of %llu\n",
-            (unsigned long long)(byte_index + string_length), (unsigned long long)bytes_max);
-        byte_index = bytes_max;
-    } else {
-        /* TODO: print the UTF8 string */
-        byte_index += (size_t)string_length;
+        fprintf(F, "    %s, Error 0x%04x, Reason length %llu\n",
+            picoquic_log_frame_names(ftype),
+            error_code, (unsigned long long)string_length);
+        if (byte_index + string_length > bytes_max) {
+            fprintf(F, "    Malformed %s, requires %llu bytes out of %llu\n",
+                picoquic_log_frame_names(ftype),
+                (unsigned long long)(byte_index + string_length), (unsigned long long)bytes_max);
+            byte_index = bytes_max;
+        } else {
+            /* TODO: print the UTF8 string */
+            byte_index += (size_t)string_length;
+        }
     }
 
     return byte_index;
 }
 
-size_t picoquic_log_application_close_frame(FILE* F, uint8_t* bytes, size_t bytes_max)
+size_t picoquic_log_connection_close_frame(FILE* F, uint8_t* bytes, size_t bytes_max)
 {
-    size_t byte_index = 1;
-    uint32_t error_code = 0;
-    uint64_t string_length = 0;
-
-    size_t l1 = 0;
-    if (bytes_max >= 4) {
-        error_code = PICOPARSE_16(bytes + byte_index);
-        byte_index += 2;
-        l1 = picoquic_varint_decode(bytes + byte_index, bytes_max - byte_index, &string_length);
-    }
-
-    if (l1 == 0) {
-        fprintf(F, "    Malformed APPLICATION CLOSE, requires %d bytes out of %d\n",
-            (int)(byte_index + picoquic_varint_skip(bytes + 3)), (int)bytes_max);
-        return bytes_max;
-    } else {
-        byte_index += l1;
-    }
+    return picoquic_log_generic_close_frame(F, bytes, bytes_max, picoquic_frame_type_connection_close);
+}
 
-    fprintf(F, "    APPLICATION CLOSE, Error 0x%04x, Reason length %d (0x%04x):\n",
-        error_code, (uint16_t)string_length, (uint16_t)string_length);
-    if (byte_index + string_length > bytes_max) {
-        fprintf(F, "    Malformed APPLICATION CLOSE, requires %d bytes out of %d\n",
-            (int)(byte_index + string_length), (int)bytes_max);
-        byte_index = bytes_max;
-    } else {
-        /* TODO: print the UTF8 string */
-        byte_index += (size_t)string_length;
-    }
+size_t picoquic_log_application_close_frame(FILE* F, uint8_t* bytes, size_t bytes_max)
```

Comment:
You may want to leave a comment that this "generic" close frame is going away (not there in Editor draft anymore).

Reply:
The "generic" name is used for the closing code common to application, connection and crypto close messages.